subsys/bluetooth/audio/ascs.c

```
@@ -3080,22 +3088,109 @@ static ssize_t ascs_cp_write(struct bt_conn *conn,
+
+static void configure_ase_char(uint8_t snk_cnt, uint8_t src_cnt)
+{
+	uint8_t snk_ases_to_rem = CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT - snk_cnt;
+	uint8_t src_ases_to_rem = CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT - src_cnt;
+	size_t attrs_to_rem;
+
+	/* Remove the Source ASEs. The ones to remove will always be at the very tail of the
+	 * attributes, so we just decrease the count withe the amount of sources we want to remove.
+	 */
+	attrs_to_rem = src_ases_to_rem * ASCS_ASE_CHAR_ATTR_COUNT;
+	ascs_svc.attr_count -= attrs_to_rem;
+
+	/* Remove the Sink ASEs.
+	 */
+	attrs_to_rem = snk_ases_to_rem * ASCS_ASE_CHAR_ATTR_COUNT;
+	if (CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT == 0 || CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	    == src_ases_to_rem) {
+		/* If there are no Source ASEs present, then we can just decrease the
+		 * attribute count
+		 */
+		ascs_svc.attr_count -= attrs_to_rem;
+	} else {
+		/* As Source ASEs are present, we need to iterate backwards (as this will likely be
+		 * the shortest distance). Find the first Sink to save, and move all Sources
+		 * backwards to it.
+		 */
+		size_t src_start_idx = ascs_svc.attr_count - (src_cnt * ASCS_ASE_CHAR_ATTR_COUNT);
+		size_t new_src_start_idx = src_start_idx - (snk_ases_to_rem *
+							    ASCS_ASE_CHAR_ATTR_COUNT);
+
+		for (size_t i = 0; i < src_cnt * ASCS_ASE_CHAR_ATTR_COUNT; i++) {
+			ascs_svc.attrs[new_src_start_idx + i] = ascs_svc.attrs[src_start_idx + i];
+		}
+
+		ascs_svc.attr_count -= attrs_to_rem;
+	}
+}
+
+int bt_ascs_register(uint8_t snk_cnt, uint8_t src_cnt)
+{
+	int err = 0;
+
+	if (ascs.registered) {
+		LOG_ERR("ASCS already registered");
+
+		return -EALREADY;
+	}
+
+	if (snk_cnt > CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT ||
+	    src_cnt > CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT) {
+		//LOG_ERR;
+		return -EINVAL;
+	}
+	configure_ase_char(snk_cnt, src_cnt);
+
+	err = bt_gatt_service_register(&ascs_svc);
+	if (err != 0) {
+		LOG_ERR("Failed to register ASCS in gatt DB");
+	}
```

```suggestion
	err = bt_gatt_service_register(&ascs_svc);
	if (err != 0) {
		LOG_DBG("Failed to register ASCS in gatt DB");
	}
```

Suggest not to use `LOG_ERR` for functions that return the error code to the application

---

Good point. Also, this should return the error code here.